core/include/scipp/core/dataset.h

```diff
@@ -963,6 +970,37 @@ SCIPP_CORE_EXPORT Dataset histogram(const Dataset &dataset, const Dim &dim);
 
 SCIPP_CORE_EXPORT Dataset merge(const DatasetConstProxy &a,
                                 const DatasetConstProxy &b);
+namespace detail {
+template <class DS, class Func>
+Dataset apply_through_dimension(const DS &ds, const Dim dimension, Func func) {
+  if (ds.containsSparse())
+    throw std::logic_error("Can't sum Dataset with sparse data");
+  if (!ds.dimensions().count(dimension))
+    throw std::logic_error("Can't sum Dataset on non existing dimension.");
```

Comment:
I would like to provide clear exception messages in here because this is what you would get working with python. I can catch the exceptions and rethrow... but here it doesn't make any sense IMHO.